### ISSUE TITLE
[requested functionality] Enable crosshair to auto-align to x-values of multiple line series.

### ISSUE THREAD

**Issue by slohmes** [uber/react-vis#988]
I have a line series with a crosshair on a single XYFlexiblePlot, that uses time x-values. When I add another line series with a different set of x-values to the plot, the crosshair only auto-aligns to one set of x-values. So the crosshair shows up for all the points in the first line series, but not the points in the second line series. When I remove the first line series, the crosshair then correctly auto-aligns to the x-values in the second line series.

I want a single crosshair to show up for all x-values in either line series. Is there currently a way to do so? If not, would it be possible to add this functionality?

Two potential ways to achieve this functionality:

1) Allow user to pass in a custom "data" prop to the Crosshair component, that is used for auto-aligning the crosshair.
This prop, if passed in, would override the results of ```getAttributeFunctor(this.props, 'x');```, which is currently used in react-vis/crosshair.js to get the x-values. In the case described above, I could pass in the combined set of points from both line series to allow the crosshair to auto-align to all points displayed by either line series.

2) Expand the getNull atrribute of the LineSeries component.
Currently, based on the boolean returned from the function passed to this attribute, the line series handles a null y-value by either displaying the point with a y-value of 0, or not displaying the line at all over that point (ie, displaying two line segements terminating at the points before and after the point with the null y-value). This attribute could be reworked to allow an additional case: if a y-value is null, do not display the point, but display a line segment from the previous point to the next point.

**Comment by mcnuttandrew** [comment]
I think you could actually do this without modifying the codebase! The way I generally approach multiple line problem mouse over problems is by making voronois.  in [this article](https://towardsdatascience.com/advanced-visualization-with-react-vis-efc5c6667b4) I give a really extensive discussion of this exact topic (I know it is real pretentious to link to one's own medium articles, but what can you do ugh).

The TLDR is that you can place voronoi points in place where you want your user to interact with them, and position so that things are just as you desire.

**Comment by mcnuttandrew** [comment]
I'm gonna close this but feel free to reopen if you like
